vmm_tests/vmm_tests/tests/tests/multiarch.rs

```diff
@@ -715,3 +716,82 @@ async fn boot_expect_fail(
 
     Ok(())
 }
+
+/// MNF guest support: capture and print recursive listing of vmbus drivers.
+#[openvmm_test(
+    openvmm_openhcl_linux_direct_x64,
+    openvmm_openhcl_uefi_x64(vhd(ubuntu_2204_server_x64))
+)]
+async fn validate_mnf_usage_in_guest(
+    config: PetriVmBuilder<OpenVmmPetriBackend>,
+) -> anyhow::Result<()> {
+    // So far, NetVSC uses MNF, StorVSC doesn't hence attach a nic to the vm.
+    let (vm, agent) = config
+        .with_vmbus_redirect(true)
+        .with_openhcl_command_line("OPENHCL_VMBUS_ENABLE_MNF=1")
+        .modify_backend(|c| c.with_nic())
+        .run()
+        .await?;
+
+    let sh = agent.unix_shell();
+    let netvsc_path = "/sys/bus/vmbus/drivers/hv_netvsc";
+
+    // List directory contents for visibility.
+    let contents = cmd!(sh, "ls -la {netvsc_path}").read().await?;
+    tracing::info!("Listing all contents of {}:\n{}", netvsc_path, contents);
+
+    // Pure helpers for parsing and path resolution.
+    fn is_hex(s: &str) -> bool {
+        !s.is_empty() && s.chars().all(|c| c.is_ascii_hexdigit())
+    }
+    fn is_guid(s: &str) -> bool {
+        let parts: Vec<&str> = s.split('-').collect();
+        matches!(parts.as_slice(), [a, b, c, d, e]
+            if [a.len(), b.len(), c.len(), d.len(), e.len()] == [8, 4, 4, 4, 12]
+            && parts.iter().all(|p| is_hex(p)))
+    }
+    fn resolve_abs(base: &str, target: &str) -> String {
+        if target.starts_with('/') {
+            return target.to_string();
+        }
+        let mut parts: Vec<&str> = base.split('/').filter(|p| !p.is_empty()).collect();
+        for seg in target.split('/') {
+            match seg {
+                "" | "." => {}
+                ".." => {
+                    parts.pop();
+                }
+                other => parts.push(other),
+            }
+        }
+        format!("/{}", parts.join("/"))
+    }
+
+    // Extract absolute target dirs from GUID-named symlink entries in ls output.
+    // Each symlink entry points to a device instance within /sys/bus/vmbus/devices/
+    // The GUIDs are the instance ID.
+    let device_dirs: Vec<String> = contents
+        .lines()
+        .filter(|line| line.starts_with('l'))
+        .filter_map(|line| {
+            let (left, target) = line.rsplit_once(" -> ")?;
+            let name = left.split_whitespace().last()?;
+            is_guid(name).then(|| resolve_abs(netvsc_path, target))
+        })
+        .collect();
+
+    tracing::info!("Devices:\n{}", device_dirs.join("\n"));
+
+    // For each device, ensure at least one monitor_id file exists.
+    for device in device_dirs {
+        let find_out = cmd!(sh, "find {device} -type f -name 'monitor_id'")
+            .read()
+            .await?;
+        let has_monitor = find_out.lines().any(|s| !s.trim().is_empty());
+        assert!(has_monitor, "no monitor_id files found in {}", device);
+    }
+
```

Comment:
Looking for some suggestions on how to validate network traffic is working as expected. My initial thought was to fetch the gateway ip address and try to ping it, but I'm not sure that approach works.

Reply:
Maybe we can open an issue for this. We could using `consomme` configured in a mode where we have a fully synthetic network, so that you could create a fake HTTP endpoint to curl or something. Although for tests like this, even just getting an IP address through consomme is probably enough to get confidence. But again, we'd want to disable access to the internet so that the tests don't accidentally rely on some external resources.